pkg/util/namer/namer.go

```diff
@@ -13,23 +13,27 @@ import (
 // an 8-character hash of the [base]-[suffix] string.  If the suffix is not too long,
 // it will truncate the base, add the hash of the base and return [base]-[hash]-[suffix]
 func GetName(base, suffix string, maxLength int) string {
+	if maxLength <= 0 {
+		return ""
+	}
 	name := fmt.Sprintf("%s-%s", base, suffix)
-	if len(name) > maxLength {
-		baseLength := maxLength - 10 /*length of -hash-*/ - len(suffix)
+	if len(name) <= maxLength {
+		return name
+	}
 
-		// if the suffix is too long, ignore it
-		if baseLength < 0 {
-			prefix := base[0:min(len(base), maxLength-9)]
-			// Calculate hash on initial base-suffix string
-			return fmt.Sprintf("%s-%s", prefix, hash(name))
-		}
+	baseLength := maxLength - 10 /*length of -hash-*/ - len(suffix)
 
-		prefix := base[0:baseLength]
+	// if the suffix is too long, ignore it
+	if baseLength < 0 {
+		prefix := base[0:min(len(base), max(0, maxLength-9))]
 		// Calculate hash on initial base-suffix string
-		return fmt.Sprintf("%s-%s-%s", prefix, hash(base), suffix)
+		shortName := fmt.Sprintf("%s-%s", prefix, hash(name))
```

Comment:
> since we use this on many places, thus we care about performance and you are already here... can you replace this with strings.Join ?

Has this surfaced on a profile?  I find `strings.Join` hard to read and this method is called right before a remote http call, so I wouldn't expect it to be high in the list.